### PR TITLE
refactor ResultSet.One() to handle nil result

### PR DESCRIPTION
**Pull request by vkagamlyk**
refactor ResultSet.One() to handle nil result

### CODE REVIEW

**Review by lyndonbauto — APPROVED** [review]
lgtm.

File: gremlin-go/driver/resultSet.go [hunk 1]
```diff
@@ -149,11 +149,15 @@ func (channelResultSet *channelResultSet) Channel() chan *Result {
 }
 
 // One returns the next Result from the channelResultSet, blocking until one is available.
```
Comment (L0Lmaker) [inline gremlin-go/driver/resultSet.go:151]: Update comment?

Comment (simonz-bq) [inline gremlin-go/driver/resultSet.go:151]: Add documentation here for the ok, as this is pretty non-standard for Go considering this method does not return a channel.

Comment (vkagamlyk) [reply gremlin-go/driver/resultSet.go:151]: added in https://github.com/apache/tinkerpop/pull/1653/commits/dcb281409d65d115fb960cc534954e84f7b5415d

Comment (vkagamlyk) [reply gremlin-go/driver/resultSet.go:151]: added in https://github.com/apache/tinkerpop/pull/1653/commits/dcb281409d65d115fb960cc534954e84f7b5415d

**Review by L0Lmaker — APPROVED** [review]

**Comment by lyndonbauto** [comment]
VOTE+1

**Comment by lyndonbauto** [comment]
> 1 week lazy consensus. Will merge.

**Pull request merged**